src/main/kotlin/net/maizegenetics/phgv2/cli/ExportPerRefRangeVCF.kt

```diff
@@ -0,0 +1,177 @@
+        .default("")
+
+    override fun run() {
+        //TODO("Not yet implemented")
+
+        val dbPath = if (dbPath.isBlank()) {
+            System.getProperty("user.dir")
+        } else {
+            dbPath
+        }
+
+        // Verify the tiledbURI - an exception is thrown from verifyURI if the URI is not valid
+        val validDB = verifyURI(dbPath, "gvcf_dataset",condaEnvPrefix)
+        runExportByRegion(dbPath, outputDir, regionsFile, samples, condaEnvPrefix, batchSize)
+
+    }
+
+    // This function will run the export command for individual regions based on user input and the regions file
+    // It should be batched at some point, but for now, it runs each region individually
+    fun runExportByRegion(dbPath: String, outputDir: String, regionsFile: String, samples: SampleProcessing,
+                          condaEnvPrefix: String, batchSize: Int) {
+        // Setup the conda environment portion of the command
+        var commandPrefix = if (condaEnvPrefix.isNotBlank()) mutableListOf("conda","run","-p",condaEnvPrefix) else mutableListOf("conda","run","-n","phgv2-conda")
+
+        val regions = convertRegionFileToRegionList(regionsFile)
+        val batches = regions.chunked(batchSize) // not sending in batches right now - will figure that out later
+
+        // THis isn't correct - we want to send a process builder command for each entry in
+        // the batch, running batch.size number of samples at a time
+
+        regions.forEach { region ->
+            // Create the data portion of the command for this batch
+
+            val outputFile = "$outputDir/export_gvcf_${region.replace(":","_")}.vcf"
+            var dataCommand = if (samples.getExportCommand()[0] == SampleFormatEnum.FILE.toString()) mutableListOf(
+                "tiledbvcf",
+                "export",
+                "--uri",
+                "$dbPath/gvcf_dataset",
+                "-m",
+                "-b",
+                "65536",
+                "-O",
+                "v",
+                "--samples-file",
+                samples.getExportCommand()[1],
+                "-o",
+                outputFile,
+                "--regions",
+                region,
+                "--tiledb-config",
+                "sm.memory_budget=10737418240,sm.memory_budget_var=21474836480,sm.skip_unary_partitioning_budget_check=true"
+            ) else mutableListOf(
+                "tiledbvcf",
+                "export",
+                "--uri",
+                "$dbPath/gvcf_dataset",
+                "-m",
+                "-b",
+                "65536",
+                "-O",
+                "v",
+                "--sample-names",
+                samples.getExportCommand()[1],
+                "-o",
+                outputFile,
+                "--regions",
+                region,
+                "--tiledb-config",
+                "sm.memory_budget=10737418240,sm.memory_budget_var=21474836480,sm.skip_unary_partitioning_budget_check=true"
```

Comment:
Maybe add some comments about the rationale behind these values?

Reply:
yes - I will add comments related to the tiledb-config options